### PR TITLE
Workflow#newActivityStub(activityInterface) method produces broken activity stubs

### PR DESCRIPTION
**Pull request by vkoby**
## What was changed:
Deprecated method that creates default ActivityOptions and added an exception if both required options aren't set.

## Why?
ActivityOptions has a requirement that either startToClose or scheduleToClose has to be defined. There's no good default value that would fit most workflows because activity would either seem like it is stuck (if it's too long) or is unable to execute in time (too short). 

Closes issue: [#413](https://github.com/temporalio/sdk-java/issues/413)

### CODE REVIEW

File: temporal-sdk/src/test/java/io/temporal/activity/ActivityOptionsTest.java [hunk 1]
```diff
@@ -41,6 +53,34 @@
...
+    TestActivity activity =
+        testEnv.newActivityStub(
+            TestActivity.class,
+            ActivityOptions.newBuilder().setScheduleToCloseTimeout(Duration.ofDays(1)).build(),
+            null);
+
+    // Check that options were set correctly
+    Map<String, Duration> optionsValues = activity.activity1();
+    Assert.assertEquals(Duration.ofDays(1), optionsValues.get("ScheduleToCloseTimeout"));
+    Assert.assertEquals(Duration.ofDays(1), optionsValues.get("StartToCloseTimeout"));
```
Comment (mfateev) [inline temporal-sdk/src/test/java/io/temporal/activity/ActivityOptionsTest.java:81]: I"m confused. Where does this timeout come from if it is not set?

File: temporal-sdk/src/test/java/io/temporal/internal/testing/ActivityTestingTest.java [hunk 2]
```diff
@@ -68,7 +75,7 @@ public String activity1(String input) {
   @Test
   public void testSuccess() {
     testEnvironment.registerActivitiesImplementations(new ActivityImpl());
-    TestActivity activity = testEnvironment.newActivityStub(TestActivity.class);
+    TestActivity activity = testEnvironment.newActivityStub(TestActivity.class, activityOptions);
```
Comment (mfateev) [inline temporal-sdk/src/test/java/io/temporal/internal/testing/ActivityTestingTest.java:78]: I don't understand why we are making these changes here. I think we should add the ability to set ActivityOptions on the default TestActivityEnvironment.

File: temporal-sdk/src/test/java/io/temporal/activity/ActivityOptionsTest.java (hunk 1 above)
Comment (vkoby) [reply temporal-sdk/src/test/java/io/temporal/activity/ActivityOptionsTest.java:81]: I wasn't sure about the reasoning behind this logic, but this is where it comes from:
https://github.com/temporalio/sdk-java/blob/4c9bdd9c445360c0b3cffb080fb8e28527910e70/temporal-sdk/src/main/java/io/temporal/activity/ActivityOptions.java#L256

File: temporal-sdk/src/test/java/io/temporal/internal/testing/ActivityTestingTest.java (hunk 2 above)
Comment (vkoby) [reply temporal-sdk/src/test/java/io/temporal/internal/testing/ActivityTestingTest.java:78]: l thought it to be misleading to have different behavior in prod and in tests. If you don't set required timeouts in the stub in prod and expect things to break, this is what should happen in tests. Consistent behavior is important for testing for all coming engineers who would have to learn the product.
Alternatively we can rename the method to something like TestActivityEnvironment#newActivityStubWithDefaults which would hopefully help with confusion.

File: temporal-sdk/src/test/java/io/temporal/activity/ActivityOptionsTest.java (hunk 1 above)
Comment (mfateev) [reply temporal-sdk/src/test/java/io/temporal/activity/ActivityOptionsTest.java:81]: That logic seems outdated. We didn't have sane defaults on the service some time ago, so it was introduced. Let's remove it from the ActivityOptions completely.

File: temporal-sdk/src/test/java/io/temporal/internal/testing/ActivityTestingTest.java (hunk 2 above)
Comment (mfateev) [reply temporal-sdk/src/test/java/io/temporal/internal/testing/ActivityTestingTest.java:78]: I think for TestActivityEnvironment we can provide default ActivityOptions with a large timeout.

File: temporal-testing/src/main/java/io/temporal/testing/TestActivityEnvironment.java [hunk 3]
```diff
@@ -39,7 +39,7 @@
  *   private static class ActivityImpl implements TestActivity {
  *    {@literal @}Override
  *     public String activity1(String input) {
- *       return Activity.getTask().getActivityType().getName() + "-" + input;
+ *       return Activity.getExecutionContext().getInfo().getActivityType().getName() + "-" + input;
```
Comment (mfateev) [inline temporal-testing/src/main/java/io/temporal/testing/TestActivityEnvironment.java:42]: I think .getName() should be removed.

**Review by mfateev — APPROVED** [review]

**Pull request merged**